pkgs/by-name/op/openbao/package.nix

```diff
@@ -0,0 +1,60 @@
+{
+  lib,
+  fetchFromGitHub,
+  buildGoModule,
+  testers,
+  openbao,
+}:
+buildGoModule rec {
+  pname = "openbao";
+  version = "2.1.0";
+
+  src = fetchFromGitHub {
+    owner = "openbao";
+    repo = "openbao";
+    rev = "v${version}";
+    hash = "sha256-QzUNb4T9uau9bWZX6ulUDyfdInGd86iClBAG72C+7mo=";
+  };
+
+  vendorHash = "sha256-Lg58NbwO7vLNRCBwJujcoVcrV018FevvdrUassnAg3k=";
+
+  proxyVendor = true;
+
+  subPackages = [ "." ];
+
+  tags = [
+    "openbao"
+    "bao"
+  ];
+
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/openbao/openbao/version.GitCommit=${src.rev}"
+    "-X github.com/openbao/openbao/version.fullVersion=${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/openbao $out/bin/bao
+  '';
```

Comment:
@wolfgangwalther @brianmay I think that's fine. As a FYI, that build script is only used in development via `make`, [the `goreleaser` script](https://github.com/openbao/openbao/blob/c3173a055a6ebcb5a2912daf4d58620eac2ba001/.goreleaser-template.yaml#L11) sets `project_name` to `bao` which eventually filters down to the binary name.

I will note `-o` isn't a ldflag, it is a compiler flag. Not sure if there's an option for that somewhere in NixOS.

Reply:
```
evaluation warning: `buildFlags`/`buildFlagsArray` are deprecated and will be removed in the 24.11 release. Use the `ldflags` and/or `tags` attributes instead of `buildFlags`/`buildFlagsArray`
```

If I try to add it to `ldflags` instead then I get errors that it can't find `a.out`.